app/components/home/index.tsx

```diff
@@ -160,6 +163,54 @@ class HomeComponent extends React.PureComponent<IHomeComponentProps, {}> {
     dispatch(Actions.toggleReadMoreBox());
   };
 
+  private shareTwitterWithComposedImage = async () => {
+    const { dispatch, homeState } = this.props;
+    const { commentInput } = homeState;
+    let fileName;
+    try {
+      const imageUrl: string = await this.drawTextAtImage(commentInput);
+      fileName = await dispatch(
+        Actions.uploadImage({
+          imageDataURL: imageUrl
+        })
+      );
+    } catch (err) {
+      console.error(err);
+    }
+
+    const plutoUrlWithImage = encodeURIComponent(`https://join.pluto.network/userImage/${fileName}`);
+
+    trackAndOpenLink(
+      `https://twitter.com/intent/tweet?text=${commentInput}&url=${plutoUrlWithImage}&hashtags=FutureOfScholComm`,
+      "signBannerTwitterShare"
+    );
+  };
+
+  private drawTextAtImage = (commentInput: string): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      try {
+        const context = this.drawingCanvas.getContext("2d");
+        const background = new Image();
+        const backgroundUrl =
+          "https://images.pexels.com/photos/104827/cat-pet-animal-domestic-104827.jpeg?h=350&auto=compress&cs=tinysrgb";
+        this.drawingCanvas.width = 600;
+        this.drawingCanvas.height = 314;
+        background.setAttribute("crossOrigin", "anonymous");
+        background.src = backgroundUrl;
+        background.onload = () => {
+          context.drawImage(background, 0, 0);
+          context.font = "20px Roboto";
```

Comment:
I guess that if the user's browser doesn't have Roboto font, the font won't be displayed as we expected

Reply:
That site use roboto font. So maybe it can work after downloading font